### PR TITLE
Refs #10970 - Fix actions katello host register

### PR DESCRIPTION
**Pull request by cpeters**
The plan_subscription_facet method was not persisting the associated
activation_keys.

### CODE REVIEW

**Comment by johnpmitsch** [comment]
LGTM @ehelms should ack because he made the initial comment

File: app/lib/actions/katello/host/register.rb [hunk 1]
```diff
@@ -108,8 +108,9 @@ def plan_subscription_facet(host, activation_keys, consumer_params)
           subscription_facet = host.subscription_facet || ::Katello::Host::SubscriptionFacet.new
           subscription_facet.host = host
           subscription_facet.update_from_consumer_attributes(consumer_params)
-          subscription_facet.save!
+          subscription_facet.save! # persist to obtain an id
```
Comment (ehelms) [inline app/lib/actions/katello/host/register.rb:111]: What about just calling `create` instead of new on line 108? Would that save a save here?

Comment (cpeters) [inline app/lib/actions/katello/host/register.rb:111]: @ehelms I tried that just now and I still see the failures. Will continue to brainstorm.

**Comment by johnpmitsch** [comment]
ACK

**Pull request merged**